daemon/README.md

```diff
@@ -0,0 +1,176 @@
+Daemon container
+================
+
+This Dockerfile may be used to bootstrap a Ceph cluster with all the Ceph daemons running.
+
+
+Usage
+-----
+
+You can use this container to bootstrap any Ceph daemon.
+
+* `CEPH_DAEMON` is the name the daemon you want to deploy (DEFAULT: none). Available values: `MON`, `OSD_DIRECTORY`, `OSD_CEPH_DISK`, `MDS`, `RGW`.
+* `CLUSTER` is the name of the cluster (DEFAULT: ceph)
+* `HOSTNAME` is the hostname of the machine  (DEFAULT: $(hostname))
+
+
+Deploy a monitor
+----------------
+
+Run:
+
+```
+$ sudo docker run -d --net=host \
+-v /etc/ceph:/etc/ceph \
+-v /var/lib/ceph/:/var/lib/ceph/ \
+-e CEPH_DAEMON=MON \
+-e MON_IP=192.168.0.20 \
+-e CEPH_PUBLIC_NETWORK=192.168.0.0/24 \
+ceph/daemon
+```
+
+List of available options:
+
+* `MON_IP`: IP address of the host running Docker
+* `MON_NAME`: name of the monitor (default to hostname)
+* `CEPH_PUBLIC_NETWORK`: CIDR of the host running Docker, it should be in the same network as the `MON_IP`
+* `CEPH_CLUSTER_NETWORK`: CIDR of a secondary interface of the host running Docker. Used for the OSD replication traffic
+
+
+Deploy an OSD
+-------------
+
+There are two available options:
+
+* use `OSD_CEPH_DISK` where you only specify a block device
+* use `OSD_DIRECTORY` where you specify an OSD mount point to your container
+
+
+### Ceph disk ###
+
+```
+$ sudo docker run -d --net=host \
+--privileged=true \
+-v /etc/ceph:/etc/ceph \
+-v /var/lib/ceph/:/var/lib/ceph/ \
+-v /dev/:/dev/ \
+-e CEPH_DAEMON=OSD \
+-e OSD_DEVICE=/dev/vdd \
+ceph/daemon
+```
+
+List of available options:
+
+* `OSD_DEVICE`:
+* `OSD_JOURNAL`:
+* `HOSTNAME`:
+
+If you do not want to use `--privileged=true`, please fall back on the second example.
+
+
+### Ceph OSD directory ###
+
+There are a number of environment variables which are used to configure
+the execution of the OSD:
+
+ -  `CLUSTER` is the name of the ceph cluster (defaults to `ceph`)
+
+If the OSD is not already created (key, configuration, OSD data), the
+following environment variables will control its creation:
+
+ -  `WEIGHT` is the of the OSD when it is added to the CRUSH map (default is `1.0`)
+ -  `JOURNAL` is the location of the journal (default is the `journal` file inside the OSD data directory)
+ -  `HOSTNAME` is the name of the host; it is used as a flag when adding the OSD to the CRUSH map
+
+The old option `OSD_ID` is now unused.  Instead, the script will scan for each directory in `/var/lib/ceph/osd` of the form `<cluster>-<osd-id>`.
+
+To create your OSDs simply run the following command:
+
+`docker exec <mon-container-id> ceph osd create`.
+
```

Comment:
Might as well continue with the instructions:

*  Create a filesystem on your OSD(s)
*  Mount your OSD(s) in to `/var/lib/ceph/osd/<cluster>-<osd-id>`
*  Execute the OSD container:  ```docker sudo docker run -d --net=host ...`

We could just merge this as is, and I can update the documentation accordingly, though.